libraries/protocol/include/steem/protocol/transaction.hpp

```diff
@@ -53,36 +55,40 @@ namespace steem { namespace protocol {
       signed_transaction( const transaction& trx = transaction() )
          : transaction(trx){}
 
-      const signature_type& sign( const private_key_type& key, const chain_id_type& chain_id );
+      const signature_type& sign( const private_key_type& key, const chain_id_type& chain_id, canonical_signature_type canon_type/* = fc::ecc::fc_canonical*/ );
 
-      signature_type sign( const private_key_type& key, const chain_id_type& chain_id )const;
+      signature_type sign( const private_key_type& key, const chain_id_type& chain_id, canonical_signature_type canon_type/* = fc::ecc::fc_canonical*/ )const;
 
       set<public_key_type> get_required_signatures(
          const chain_id_type& chain_id,
          const flat_set<public_key_type>& available_keys,
          const authority_getter& get_active,
          const authority_getter& get_owner,
          const authority_getter& get_posting,
-         uint32_t max_recursion = STEEM_MAX_SIG_CHECK_DEPTH
+         uint32_t max_recursion/* = STEEM_MAX_SIG_CHECK_DEPTH*/,
```

Comment:
I left the defaults there because there is some cleanup that needs occur post HF that will cleanup this interface and restore it to its pre hardfork state.